nixops/deployment.py

```diff
@@ -439,6 +439,9 @@ def evaluate(self) -> None:
 
         # Extract info about other kinds of resources.
         for res_type, cfg in config["resources"].items():
+            # skip problematic resources entries
+            if res_type in ["deployment", "_name", "_type"]:
```

Comment:
> @roberth what is about that `mypy-ratchet` failing test?

Apparently this new comparison doesn't have precise types. I don't think that's a big deal in itself.

I don't know where `_name` and `_type` are from, but I think you've added `deployment` through `deploymentInfoModule`. Those resource defaults can only realistically be used to set things in `_module`, because anything else will pollute actual resource parameters.
Setting things like aws defaults is enticing, but problematic and potentially confusing when later adding a non-aws resource.
`_module.args` doesn't seem that useful when resources can reference higher scopes including the whole network. You might as well reference those configs directly.
So I think the only use case for the resource defaults option is legacy compatibility.